boa/src/builtins/array/mod.rs

```diff
@@ -58,7 +58,7 @@ impl BuiltIn for Array {
         )
         .name(Self::NAME)
         .length(Self::LENGTH)
-        .property("length", 0, Attribute::all())
+        .property("length", 0, Attribute::WRITABLE | Attribute::NON_ENUMERABLE)
```

Comment:
I would add `Attribute::PERMANENT` too, to make it more explicit, even if it's the same.